### PR TITLE
Fix for upstream changes

### PR DESCRIPTION
**Pull request by chris-morgan**
I *think* these are things that came since the latest nightly while #38 has the stuff in the most recent nightly build, but I’m not certain.

Anyway, #38 + #39 + this is enough to get it all to compile for me on my very recent master rustc build. Mish it up, mash it up, rebase it, gut it, ignore it—do whatever you like.

### CODE REVIEW

File: phf/tests/test.rs [hunk 1]
```diff
@@ -112,7 +112,7 @@ mod map {
         static MAP: phf::Map<&'static str, int> = phf_map!(
             "a" => 0,
         );
-        assert_eq!(Some(&0), MAP.get("a".to_string()[]));
```
Comment (sfackler) [inline phf/tests/test.rs]: I don't think these changes should be necessary.

**Comment by sfackler** [comment]
Needs a rebase

File: phf/tests/test.rs (hunk 1 above)
Comment (chris-morgan) [inline phf/tests/test.rs]: Parallel changes were made upstream and it was required to get it to compile. I dunno if it’s temporary.

**Comment by chris-morgan** [comment]
Rebased.

**Pull request closed without merging**